storage/kafka/src/main/java/io/apicurio/registry/kafka/KafkaRegistryStorage.java

```diff
@@ -48,102 +63,405 @@
 @ApplicationScoped
 public class KafkaRegistryStorage extends SimpleMapRegistryStorage implements KafkaRegistryStorageHandle {
 
+    private static final Logger log = LoggerFactory.getLogger(KafkaRegistryStorage.class);
+
+    @ConfigProperty(name = "registry.kafka.snapshot.requests", defaultValue = "1000")
+    long snapshotRequests;
+
+    @ConfigProperty(name = "registry.kafka.snapshot.period", defaultValue = "1200") // 2 days
+    long snapshotPeriod; // the time in minutes from the last time we did snapshot
+
+    @ConfigProperty(name = "registry.kafka.snapshot.topic", defaultValue = "snapshot-topic")
+    String snapshotTopic;
+
+    @ConfigProperty(name = "registry.kafka.schedule.period", defaultValue = "1")
+    long schedulePeriod; // schedule check period in minutes
+
+    @ConfigProperty(name = "registry.kafka.registry.topic", defaultValue = "registry-topic")
+    String registryTopic;
+
+    @Inject
+    ProducerActions<Reg.UUID, Reg.RegistryValue> registryProducer;
+
     @Inject
-    Function<ProducerRecord<Reg.UUID, Reg.SchemaValue>, CompletableFuture<RecordMetadata>> schemaProducer;
+    ProducerActions<Long, StorageSnapshot> snapshotProducer;
 
     private volatile long offset = 0;
 
-    private final Map<UUID, CompletableFuture<Object>> cfMap = new ConcurrentHashMap<>();
+    private final Map<UUID, CF<Object>> cfMap = new ConcurrentHashMap<>();
```

Comment:
`cfMap` is too cryptic for my taste. How about something like `outstandingRequests`?